google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -333,6 +341,11 @@ def publish(self, topic, data, ordering_key="", **attrs):
 
             pubsub_v1.publisher.exceptions.MessageTooLargeError: If publishing
                 the ``message`` would exceed the max size limit on the backend.
+
+            :exception:`~pubsub_v1.publisher.exceptions.FlowControlLimitError`:
```

Comment:
In Java, we didn't throw an exception, we completed the future with the exception. Perhaps we should do the same here for consistency.

Reply:
I'll see if that can be implemented without violating the boundaries too much. Maybe by `add()` also accepting the publish future, and resolving it with an exception rather than raising the exception directly...

... and there you go, answered it to myself. :smile:

Reply:
Indeed we should, thanks. We do not not include errors raised through the publish futures.